File: src/main/java/org/leonard/jthesaurus/views/HomeView.java
package org.leonard.jthesaurus.views;

import javax.swing.*;
import java.awt.*;

import static javax.swing.SpringLayout.*;
import static org.leonard.jthesaurus.Application.vh;
import static org.leonard.jthesaurus.Application.vw;

public class HomeView extends JPanel {

    TextField textField;
    JButton textButton;

    double inlineMargin = 0.3;

    int componentHeight = 32;
    int componentWidth = 160;

    SpringLayout springLayout;

    public HomeView() {
        springLayout = new SpringLayout();
        setLayout(springLayout);


        textField = new TextField();
        this.setPreferredSize(new Dimension(componentWidth, componentHeight));
        this.setMaximumSize(new Dimension(componentWidth, componentHeight));
        this.springLayout.putConstraint(WEST, textField, vw(inlineMargin), WEST, this);
        this.springLayout.putConstraint(EAST, textField, -vw(inlineMargin), EAST, this);
        this.springLayout.putConstraint(SOUTH, textField, -vh(0.465), SOUTH, this);
        this.springLayout.putConstraint(NORTH, textField, vh(0.465), NORTH, this);
        this.add(textField);
        this.repaint();

        textButton = new JButton("\uD83D\uDD0E");
        this.textButton.setPreferredSize(new Dimension(80   , 40));
        this.textButton.setMaximumSize(this.textButton.preferredSize());
        this.textButton.setFont(this.textButton.getFont().deriveFont(16f));
        //this.textButton.addActionListener();
        this.add(textButton);

    }


}

File: src/main/kotlin/org/leonard/jthesaurus/view/HomeView.kt
package org.leonard.jthesaurus.view

import org.leonard.jthesaurus.Application
import org.leonard.jthesaurus.Application.vh
import org.leonard.jthesaurus.Application.vw
import java.awt.Dimension
import javax.swing.JPanel
import javax.swing.JTextField
import javax.swing.SpringLayout
import javax.swing.SpringLayout.EAST
import javax.swing.SpringLayout.WEST

object HomeView: JPanel() {
    private val layout: SpringLayout
    private val searchField: JTextField

    init {
        this.preferredSize = Application.size
        this.layout = SpringLayout()
        this.setLayout(layout)

        /* TextField */
        this.searchField = JTextField()
        this.searchField.preferredSize = Dimension(320, 40)
        this.searchField.maximumSize = this.searchField.preferredSize

        /* TextField Layout */
        this.layout.putConstraint(WEST, searchField, 0.3.vw(), WEST, this)
        this.layout.putConstraint(EAST, searchField, (-0.3).vw(), EAST, this)
        this.layout.putConstraint(SpringLayout.NORTH, searchField, 0.465.vh(), SpringLayout.NORTH, this)
        this.layout.putConstraint(SpringLayout.SOUTH, searchField, (-0.465).vh(), SpringLayout.SOUTH, this)

        /* TextField Finalization */
        this.add(this.searchField)
    }
}